### PR TITLE
Improve imports/exports to reduce the likelihood of a conflict

### LINKED ISSUE #498: Importing CTFd-exported ZIP gives DateTime error
**Environment**:

  - CTFd Version/Commit: master
  - Operating System: Debian Testing
  - Web Browser and Version: N/A

**What happened?**
In trying to import a ZIP file that was previously exported with CTFd, an error is thrown stating "SQLite DateTime type only accepts Python"

**What did you expect to happen?**
I expected the data to import successfully

**How to reproduce your issue**
sudo python import.py ~/CTF-DATA.zip

**Any associated stack traces or error logs**
```
> sudo python import.py ~/CTF-DATA.zip 
/*\ CTFd has updated and must update the database! /*\
/*\ Please backup your database before proceeding! /*\
/*\ CTFd maintainers are not responsible for any data loss! /*\
Run database migrations (Y/N)y
 * Loaded module, <module 'CTFd.plugins.challenges' from '/var/www/ctfd/CTFd/plugins/challenges/__init__.pyc'>
 * Loaded module, <module 'CTFd.plugins.keys' from '/var/www/ctfd/CTFd/plugins/keys/__init__.pyc'>
Traceback (most recent call last):
  File "import.py", line 9, in <module>
    import_ctf(sys.argv[1])
  File "/var/www/ctfd/CTFd/utils.py", line 802, in import_ctf
    table.insert(entry)
  File "/usr/local/lib/python2.7/dist-packages/dataset/table.py", line 80, in insert
    res = self.db.executable.execute(self.table.insert(row))
  File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/engine/base.py", line 945, in execute
    return meth(self, multiparams, params)
  File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/sql/elements.py", line 263, in _execute_on_connection
    return connection._execute_clauseelement(self, multiparams, params)
  File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/engine/base.py", line 1053, in _execute_clauseelement
    compiled_sql, distilled_params
  File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/engine/base.py", line 1121, in _execute_context
    None, None)
  File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/engine/base.py", line 1402, in _handle_dbapi_exception
    exc_info
  File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/util/compat.py", line 203, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb, cause=cause)
  File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/engine/base.py", line 1116, in _execute_context
    context = constructor(dialect, self, conn, *args)
  File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/engine/default.py", line 623, in _init_compiled
    param.append(processors[key](compiled_params[key]))
  File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/dialects/sqlite/base.py", line 610, in process
    raise TypeError("SQLite DateTime type only accepts Python "
sqlalchemy.exc.StatementError: (exceptions.TypeError) SQLite DateTime type only accepts Python datetime and date objects as input. [SQL: u'INSERT INTO teams (name, email, password, website, affiliation, country, bracket, banned, verified, admin, joined) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)']
```

### PR DESCRIPTION
**Pull request by ColdHeat**
Closes #498 by checking for multiple date formats
Resolves some import conflicts with IDs not matching up and conflicts with users having the same team name. 
All exports should include their alembic_version or else we're just guessing when the export came from.

### CODE REVIEW

File: CTFd/utils.py [hunk 1]
```diff
@@ -888,11 +903,31 @@ def import_ctf(backup, segments=None, erase=False):
...
-                    table.insert(entry)
+                                    continue
+                    for k, v in entry.items():
+                        if k == 'chal' or k == 'chalid':
+                            entry[k] += chals_base
+                        if k == 'team' or k == 'teamid':
+                            entry[k] += teams_base
+
+                    if item == 'teams':
+                        table.insert_ignore(entry, ['email'])
```
Comment (dbaxa) [inline CTFd/utils.py]: Doesn't this mean you can end up with a duplicate team import (the email will match) ?

**Comment by ColdHeat** [comment]
I tried a couple of imports of the same data and the behavior was as I
expected. Challenges were repeated but teams were not which was the
behavior I was after.

It doesn't insert teams with emails that already exist from my analysis.

I think the dataset documentation may be a little unclear but INSERT IGNORE
in MySQL makes sense.

However I think it is possible to have two teams with the same name but
different emails.

**Comment by dbaxa** [comment]
Fair enough.

**Comment by ColdHeat** [comment]
If you find a conflict in your imports or think this won't work just post
the test case and I'll take a look :)

**Pull request merged**